packages/api/src/main/messaging/MessagingClient.ts

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import amqp from 'amqplib'
+import { v4 as uuidv4 } from 'uuid'
+
+import { EventMessage, QueryMessage } from './config'
+import { MESSAGE_QUEUE_URL } from './constants'
+
+const RANDOM_QUEUE_NAME = ''
+const DEFAULT_ROUTING_KEY = ''
+const CONNECT_RETRY_INTERVAL = 5000
+
+// TODO: docs
+
+type ReplyCallbacks = Map<string, (value: any) => void>
+
+class MessagingClient {
+  private static instance?: MessagingClient
+
+  /**
+   * Obtain a singleton instance of a MessagingClient used to send messages.
+   *
+   * @returns The singleton.
+   */
+  public static getInstance(): MessagingClient {
+    if (!MessagingClient.instance) {
+      MessagingClient.instance = new MessagingClient()
+    }
+
+    return MessagingClient.instance
+  }
+
+  private connection?: amqp.Connection
+  private channel?: amqp.ConfirmChannel
+  private replyQueue?: amqp.Replies.AssertQueue
+  private replyCallbacks: ReplyCallbacks
+
+  private constructor() {
+    this.replyCallbacks = new Map()
+  }
+
+  public async connect(retry = true): Promise<amqp.ConfirmChannel> {
```

Comment:
should we add Jsdocs to the public functions here?

Reply:
I did a docs run through all of these PRs. Definitely let me know if you have any comments on them or things you think I should add/change

Reply:
LGTM!